src/mvs_eland/E1_process_results.py

```diff
@@ -254,75 +260,155 @@ def get_results(settings, bus_data, dict_asset):
+        for bus_instance in bus_name:
+            flow_tuple = get_touple_for_oemof_results(
+                dict_asset[LABEL], asset_group, bus_instance
+            )
+            # Get flow information
             get_flow(
-                settings,
-                bus_data[output_name],
-                dict_asset,
-                output_name,
-                direction="output",
+                settings=settings,
+                bus=bus_data[bus_instance],
+                dict_asset=dict_asset,
+                flow_tuple=flow_tuple,
             )
-        else:
-            for bus in output_name:
-                get_flow(settings, bus_data[bus], dict_asset, bus, direction="output")
+            # Get capacity information
+            get_optimal_cap(bus_data[bus_instance], dict_asset, flow_tuple)
 
-    # definie capacities. Check if the component has multiple input or output busses
-    if OUTPUT_BUS_NAME in dict_asset and INPUT_BUS_NAME in dict_asset:
-        if not isinstance(output_name, list):
-            get_optimal_cap(bus_data[output_name], dict_asset, output_name, "output")
-        else:
-            for bus in output_name:
-                get_optimal_cap(bus_data[bus], dict_asset, bus, "output")
+    return
 
-    elif INPUT_BUS_NAME in dict_asset:
-        if not isinstance(input_name, list):
-            get_optimal_cap(bus_data[input_name], dict_asset, input_name, "input")
-        else:
-            for bus in input_name:
-                get_optimal_cap(bus_data[bus], dict_asset, bus, "input")
 
-    elif OUTPUT_BUS_NAME in dict_asset:
-        if not isinstance(output_name, list):
-            get_optimal_cap(bus_data[output_name], dict_asset, output_name, "output")
-        else:
-            for bus in output_name:
-                get_optimal_cap(bus_data[bus], dict_asset, bus, "output")
-    return
+def get_parameter_to_be_evaluated_from_oemof_results(asset_group, asset_label):
+    r"""
+    Determines the parameter that needs to be evaluated to determine an asset`s optimized flow and capacity.
+
+    Parameters
+    ----------
+    asset_group: str
+        Group that asset belongs to
+
+    asset_label: str
+        Label of the asset, needed for log message
+
+    Returns
+    -------
+    parameter_to_be_evaluated: str
+        Parameter that will be processed to get the dispatch and capacity of an asset
+
+    Notes
+    -----
+    Tested by:
+    - test_get_parameter_to_be_evaluated_from_oemof_results()
+    """
+    if asset_group in asset_groups_defined_by_influx:
+        parameter_to_be_evaluated = INPUT_BUS_NAME
+
+    elif asset_group in asset_groups_defined_by_outflux:
+        parameter_to_be_evaluated = OUTPUT_BUS_NAME
+
+    else:
+        logging.warning(
+            f"The asset {asset_label} is of group {asset_group}, which is not defined in E1.get_results()."
+        )
+
+    return parameter_to_be_evaluated
+
+
+def get_touple_for_oemof_results(asset_label, asset_group, bus):
```

Comment:
```suggestion
def get_tuple_for_oemof_results(asset_label, asset_group, bus):
```

Reply:
Maybe the functions `set_nested_value`, `get_nested_value` and `split_nested_path` which are located within `test_sensitivity.py` could be useful? We could move them to `mvs_eland.utils`

Reply:
I will have to look at them. I will make this into an issue, so that we remember to look at it later. Changing working code is something lower on my priority-list ;)